### ISSUE TITLE
[BITV] 9.3.3.2/3.3 - Comment section: The input fields for creating and editing comments lack visual labels. (2)

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/server#37087]
see also https://github.com/nextcloud/server/issues/37091

The important note "Write message, use "@" to mention someone, use ":" for emoji autocompletion …," is only available as a placeholder element and should ideally be placed as visual text below the field linked to it via aria-describedby attribute. 

<details>
<summary>Details</summary>

https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-9f92015c98-v3-n3

</details>

- [ ] 9\.3.3.2/10.3 - https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-9f92015c98-v10-n3

**Comment by dregad** [comment]
While you're at it maybe you'll consider that the placeholder text is too long, and overflows the comment field, overlapping other text below as shown in the screenshot

![image](https://github.com/nextcloud/server/assets/449891/847755d5-3765-4f0e-a9c3-fd935e8252b6)

**Comment by AndyScherzinger** [comment]
This issue would resolve what has been commented on this issue... If the placeholder text gets moved below the input field than the issue with the length is automatically solved.

**Comment by AndyScherzinger** [comment]
@ShGKme maybe a starter issue for you?

**Comment by AndyScherzinger** [comment]
@jancborchardt what should be used as a placeholder when the text is moved below the field as a description?

**Comment by ShGKme** [comment]
@michaelnissenbaum

Is that correct that [`aria-placeholder`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-placeholder) can be used here instead of `aria-describedby` for the placeholder?

**Comment by ShGKme** [comment]
> @jancborchardt what should be used as a placeholder when the text is moved below the field as a description?

What about:
1. Make the default placeholder short (see https://github.com/nextcloud-libraries/nextcloud-vue/pull/4408)
2. Hide a long one with an ellipsis (a part will be cut)

**Comment by michaelnissenbaum** [comment]
@ShGKme `aria-placeholder` is not a substitute for `aria-describedby`. They are used for different purposes. In any case, please ensure that placeholders are not used for conveying important information. They often have poor contrast, disappear upon user input, and might not be announced by screen readers. It's recommended to place important additional information, such as field format or input instructions, visibly below the input field and associate them with the field using `aria-describedby`.

**Comment by ShGKme** [comment]
> Can someone add an image showing how the solution will look like?

@rakekniven see https://github.com/nextcloud-libraries/nextcloud-vue/pull/4482

**Comment by marcoambrosini** [comment]
I think here we should use the same label that we've added to text fields and cut it short to "Write a comment" for its text.

I would avoid this long suggestion altogether in such small inputs. It uses a lot of space and becomes redundant the moment any given user first learns about the functionality. This belongs in the documentation or in an informative tooltip that pops up the first time you use the feature.

Going forward there will be more and more things that one will be able to do with an input field, just as important or more important than emojis and mentions depending on the context. We cannot write a docs page in there :)

**Comment by ShGKme** [comment]
> I would avoid this long suggestion altogether in such small inputs.

We can make the default description short and initial size of the text field 2 lines. For example, in this case shorter to:
```
"@" to mention, ":" for emoji
```

But it still can be too long to be 1-line in some languages and small screens.

> or in an informative tooltip that pops up the first time you use the feature

I'd avoid this approach as too complex for that. If we don't want to show the tooltip each time user re-login, we need to store it in the database, grab on each page rendering with initial state...

> This belongs in the documentation

Then we need to mention it in the all places with rich contenteditable. Currently there is no documentation about commenting files at all in the user documentation.

I think, UI should be understandable without reading all the documentation, especially speaking about accessibility.

---

As an alternative I may propose adding a small question mark button that opens a tooltip with any descriptions and/or links. But it must be 44px anyway... Still I have no idea where to put it.

Current alternative: https://github.com/nextcloud-libraries/nextcloud-vue/pull/4408

**Comment by ShGKme** [comment]
### Option 1: add visible description block

- ✅ Supports custom content
- ✅ Visible and accessible
- ❌ Large, multiline for long instructions / small layout

PR: https://github.com/nextcloud-libraries/nextcloud-vue/pull/4482

![image](https://github.com/nextcloud/server/assets/25978914/f517d2da-7e50-4c36-88b1-bc3806b2ac44)

![image](https://github.com/nextcloud/server/assets/25978914/8dbe4bbb-dbf9-4185-b0bc-d3d2c79afced)


### Option 2: remove instructions at all (move to the user documentation)

- ✅ Clean
- ✅ Fits any layout :D 
- ❌ Supposes that users have read documentation or just know about the feature

![image](https://github.com/nextcloud/server/assets/25978914/fab39a64-803c-4ade-b437-efa99cd9c8c1)

### Option 3: add one-time instruction

- ✅ Clean
- ✅ Fits any layout :D 
- ✅ Yet understandable
- ❌ Complex implementation with backend involved or is shown on every login
- ❌ What to do if a user has forgot the instructions?

### Option 4: add a help button to show instructions

- ✅ Not so large
- ✅ Accessible
- ❔ May not fit any layout

(design from paint)

![image](https://github.com/nextcloud/server/assets/25978914/81ecb307-b59a-40e2-b275-64eea5822f38)
